### ISSUE TITLE
Status of BLE?

### ISSUE THREAD

**Issue by thmichel** [jdrescher2006/Laufhelden#123]
Hi,

I know Laufhelden does not support BLE devices yet. Would really love to see this implemented, I own a Jabra Sport Headset with Heart Rate and Motion sensor. 

Is there a chance that BLE will be supported any time soon? Happy to donate to raise the motivation a bit :-)

My coding skills in C++ are a bit outdated so I am not of much help there but would be happy to test any experimental versions.

Best regards and thanks for this great app!

Thomas.

**Comment by ncartron** [comment]
Same here, using my Polar HRM which does BLE would be nice.
Happy to fund a little :)

**Comment by jdrescher2006** [comment]
Hi guys,
some time ago I heard that someone is working on a BLE heartrate app for Sailfish. Since it didn't appear yet, maybe it's time to do something about it.
There are a lot of ways to start with this, for example take a look at the "Bluetooth Low Energy Heart Rate Game" QT example.
The problem is I am quite consumed at the moment and can't really concentrate on developing for SFOS. I hope that will change someday soon.

**Comment by thmichel** [comment]
Hi,

I did have a look at your code and on the Bluetooth Low Energy scanner example. That looked pretty straigth forward. The difference in connecting to a BLE device seems to be that you need a QBluetoothDeviceInfo object as parameter, in your code you only store the name and MAC address when scanning. So I think first it would be necessary to save a list of the QBluetoothDeviceInfo onbject of all scanned devices so they can be used for connecting. Then depending on the QBluetoothDeviceInfo::LowEnergyCoreConfiguration Flag the correct connection method could be used.

I am not very familiar with github, forking of code etc. but downloaded your code as ZIP file and will try to experiment a bit with this if you don't mind. If I am successfull I could send you the code via mail.

**Comment by jdrescher2006** [comment]
You are right, first you need to scan for BLE devices. Then the user chooses the one from the list of found devices.
The Bluetooth Low Energy scanner example would be very useful. Do you have the code of it?

**Comment by jdrescher2006** [comment]
This sounds very good! To keep it more simple for you, you could do your work in a small separate project. Then I could do the integration work into Laufhelden later.

**Comment by thmichel** [comment]
Yes, currentlx I am doing it completely offline as I am not very expirienced with forkin on github. If you send me an email to tom [at] michel.ruhr I can send you the code when I am satisfied with it. 

@ncartron: I could send you the rpm for testing wiht you Polar device so I can see if it is working with different devices too as I have only one device to test with.

I made some progress already, connection to the device seems to be more stable now - had it running for about 10 minutes yesterday without problems.

**Comment by ncartron** [comment]
Hi,

Yes please do, i can test it with my Polar HRM.

Thanks! 
Nico

On Saturday, 24 November 2018, thmichel wrote:
> Yes, currentlx I am doing it completely offline as I am not very expirienced with forkin on github. If you send me an email to tom [at] michel.ruhr I can send you the code when I am satisfied with it. 
> 
> @ncartron: I could send you the rpm for testing wiht you Polar device so I can see if it is working with different devices too as I have only one device to test with.
> 
> I made some progress already, connection to the device seems to be more stable now - had it running for about 10 minutes yesterday without problems.
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/jdrescher2006/Laufhelden/issues/123#issuecomment-44135610

-- 
Sent from my Sailfish device

**Comment by thmichel** [comment]
Nico,

I would need your email address then. Just send me a mail to the address mentioned above and I'll send you the packet.

**Comment by ncartron** [comment]
Hi Tom, just did it!

CHeers,
Nico

**Comment by jdrescher2006** [comment]
Hello Tom,
I have a BLE heart rate belt and did some tests with your software The search finds it (SP-HRM-BLE-200) but when I try to connect there is a problem:
`Device::scanServices:139 - Trying to connect to  "EF:44:35:C8:10:B9"
Device::scanServices:157 - Device to connect not found!
Device::scanServices:162 - Not a valid device`

Maybe it is not identified as heart rate device?

**Comment by thmichel** [comment]
Hi,

Looking at the logs it seems that the divice you are trying to connect is not yet discovered. I know this happens if you connect to a device without hitting „Scan for devices“ first. It is something I Need to fix, not sure how yet.

Can you try to scan for devices first? Once the scan is finished, it should appear in the list and you should be able to connect.

You could send me a full console Output if you like, then I could check. Another possibility is that this device is using random addresses (from what I read a kind of privacy fundtion) so the mac address might not be valid any more.

In the device.cpp file this is set to fals in the device Initiator:

Device::Device():
    connected(false), controller(0), m_deviceScanState(false), randomAddress(false)

You might try to set this to true, recompile it and test again. If that fixes it, I could try to see if there is a way to automatically detect this.

Regards,

Tom.


________________________________
Von: Jens Drescher <notifications@github.com>
Gesendet: Monday, November 26, 2018 8:13:46 PM
An: jdrescher2006/Laufhelden
Cc: Thomas Michel; Author
Betreff: Re: [jdrescher2006/Laufhelden] Status of BLE? (#123)


Hello Tom,
I have a BLE heart rate belt and did some tests with your software The search finds it (SP-HRM-BLE-200) but when I try to connect there is a problem:
Device::scanServices:139 - Trying to connect to "EF:44:35:C8:10:B9" Device::scanServices:157 - Device to connect not found! Device::scanServices:162 - Not a valid device

Maybe it is not identified as heart rate device?

**Comment by jdrescher2006** [comment]
Hi Tom,

if I scan first and then press the connect button, I get the following:
`onStatusChanged:19 - BTPage
[D] Device::scanServices:139 - Trying to connect to  "EF:44:35:C8:10:B9"
[D] Device::scanServices:157 - Device to connect not found!
[W] Device::scanServices:162 - Not a valid device
[D] Device::startDeviceDiscovery:93 - Scanning for devices ...
[D] Device::addDevice:106 - "Last device added: HXM037426"
[D] expression for onDeviceFound:46 - 1
[D] Device::addDevice:106 - "Last device added: 4D-9E-BA-FF-12-19"
[D] expression for onDeviceFound:46 - 2
[D] Device::addDevice:106 - "Last device added: Pebble 0797"
[D] expression for onDeviceFound:46 - 3
[D] Device::addDevice:106 - "Last device added: SP-HRM-BLE-200"
[D] expression for onDeviceFound:46 - 4
[D] Device::deviceScanFinished:119 - Scan finished!
[W] unknown:207 - file:///usr/share/harbour-laufhelden/qml/pages/BTConnectPage.qml:207: ReferenceError: id_BluetoothData is not defined
[D] Device::scanServices:139 - Trying to connect to  "EF:44:35:C8:10:B9"
[D] Device::scanServices:178 - Trying to connect to  "SP-HRM-BLE-200"
[D] Device::scanServices:192 - Cannot connect to remote device.  QLowEnergyController::Error(UnknownError)
[W] unknown:0 - void QBluetoothSocketPrivate::_q_readNotify() 41 error: -1 "Resource temporarily unavailable"
[D] Device::scanServices:192 - Cannot connect to remote device.  QLowEnergyController::Error(UnknownError)
[W] Device::deviceDisconnected:297 - Disconnect from device
[D] Device::scanServices:139 - Trying to connect to  "EF:44:35:C8:10:B9"
[D] Device::scanServices:192 - Cannot connect to remote device.  QLowEnergyController::Error(UnknownError)
[W] unknown:0 - void QBluetoothSocketPrivate::_q_readNotify() 41 error: -1 "Resource temporarily unavailable"
[D] Device::scanServices:192 - Cannot connect to remote device.  QLowEnergyController::Error(UnknownError)
[W] Device::deviceDisconnected:297 - Disconnect from device
`

**Comment by thmichel** [comment]
I checked with my device, I get the same error when I try to connect using random address method to it - so I guess this is the issue.
I have published a new rpm that first tries to connect using random address and in case of an error switches to public address. Drawback is that ther is a 30 sec. timeout before it tries to cennect using public addresses.
I think some status needs to be shown to the user about what's going on. 
We now have 3 connection methods: BLE public, BLE random or classic Bluetooth.
It's not very nice to have to cycle through this, I need to think on how the could be improved. Probably a setting could be stored for the last used device so that at least this one connects quicker. Sadly I have not yet find a way to find out wich connection method is correct except to try one after the other.

**Comment by jdrescher2006** [comment]
The battery service does not seem to work. How can it be fixed?

What we also need is a better integration into Laufhelden.
I think the user should be able to choose between "Bluetooth Classic" and "Bluetooth LE". This could be done via a combobox in the BTConnectPage.qml.
The Meerun Android App did this exactly like that.

The chosen Bluetooth technology must then be saved as parameter in the settings.

**Comment by thmichel** [comment]
Yes, it's far from being perfect ;-)

First wanted to get it up and running to see if it works at all. Regarding the combo box, was thinking about this as well. Drawback is that the user needs to _know_ what type of device he has (and I'd rather not rely on that :-) ).

I would like to first experiment with other options, e.g. trying all three connection methods simultaneously (if that's possible at all). I hope I find a way by the end of next week, otherwise the combo box can be used.

Regarding Battery Service: Can't you see a value at all or is it just not changing? It works for my device so it should be possible to fix this, I will add some more debug commands so we can see what's going on. You will see a log entry only when the level changes, not every second like the hart beat. I'll probably remove the heartbeat log now that it works to reduce the output a bit.

**Comment by jdrescher2006** [comment]
OK take your time!

I will do a few more tests with the battery service and report back here!

**Comment by thmichel** [comment]
Hi Nico,

I just double-checked, it can be found here:
https://github.com/thmichel/Laufhelden/releases

Regards,
Tom.

**Comment by ncartron** [comment]
Right, so that's the page I was checking, and for some reason it was not here
Now I have it, thanks! :)

On 29-Nov-2018 13:35 CET, <notifications@github.com> wrote:

> Hi Nico,
> 
> I just double-checked, it can be found here:
> https://github.com/thmichel/Laufhelden/releases
> 
> Regards,
> Tom.
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/jdrescher2006/Laufhelden/issues/123#issuecomment-442817957

-- 
Nico

**Comment by ncartron** [comment]
Now installed, I'll give it a try tomorrow and will let you know Tom!

**Comment by ncartron** [comment]
Right, so I tried this morning quickly with my Polar HRM, and it seems to work! (see pica)
Battery level doesn't work though, but i don't really mind, and I'm not even sure the official Polar app does provide this info anyway.
!

[20181130083214](https://user-images.githubusercontent.com/22927685/49276250-40803200-f47e-11e8-955c-819ce9252f6f.jpg)

**Comment by thmichel** [comment]
Hi,

I have created  a new release on my fork. As parallel connections did not work, I created a corresponding setting on the Bluetooth connection page. 
I have also re-added the support for "classic" bluetooth devices, but can't test as I don't own one.
The BLE stuff seems to be quite stable now (at least for my device). 

Please have a look at it and see whether we can push this back to your code.

**Comment by jdrescher2006** [comment]
OK, I will check it out. I am looking forward to do this on monday.

**Comment by jdrescher2006** [comment]
Did some tests with Bluetooth classic. Does not seem to work anymore.
Here is the output:
[W] unknown:46 - file:///usr/share/harbour-laufhelden/qml/pages/BTConnectPage.qml:46:5: QML Connections: Cannot assign to non-existent property "onConnecting"
[D] Device::startDeviceDiscovery:105 - Scanning for devices ...
[D] Device::addDevice:128 - "Last device added: HXM037426"
[D] expression for onDeviceFound:55 - 1
[D] Device::addDevice:128 - "Last device added: SP-HRM-BLE-200"
[D] expression for onDeviceFound:55 - 2
[D] onStatusChanged:27 - BTPage
[D] Device::deviceScanFinished:141 - Scan finished!
[D] Device::scanServices:157 - Trying to connect to  "00:07:80:0F:AC:95"
[D] Device::scanServices:193 - Trying to connect to address  "HXM037426"
[W] unknown:0 - void QBluetoothSocketPrivate::_q_readNotify() 50 error: -1 "Resource temporarily unavailable"
[W] Device::deviceDisconnected:346 - Disconnect from device

**Comment by jdrescher2006** [comment]
Maybe we can keep the old implementation as it was in Laufhelden. And if you choose Bluetooth Classic, we would use those old methods. Then if the user chooses BLE (random or public), we would use your new cpp classes.
If you like, I can prepare this and add it to Laufhelden. I might find some time for that next week.

**Comment by thmichel** [comment]
That sure would be an option. I thought it might be easier to have both types in one class (so only one model is needed in the qml page). I published a new release for testing as I set a wrong port in the connectToServie call. I used 0 (NotOpen), you used 1 (ReadOnly). I have corrected this.

Another issue could be that I allowed to connect to the device while still scanning - that works for BLE devices, not sure about others. Maybe you would have to wait until the scan finishes.

Feel free to test again (or switch to two methods).

**Comment by jdrescher2006** [comment]
Still doesn't work, scanning was finished:
[D] Device::startDeviceDiscovery:105 - Scanning for devices ...
[D] Device::addDevice:128 - "Last device added: SP-HRM-BLE-200"
[D] expression for onDeviceFound:55 - 1
[D] Device::addDevice:128 - "Last device added: 78-B3-F7-17-3F-22"
[D] expression for onDeviceFound:55 - 2

[D] onStatusChanged:27 - BTPage
[D] Device::addDevice:128 - "Last device added: f023c2XO"
[D] expression for onDeviceFound:55 - 3

[D] Device::addDevice:128 - "Last device added: f022c2YZ"
[D] expression for onDeviceFound:55 - 4
[D] Device::addDevice:128 - "Last device added: 45-7F-03-B9-65-E8"
[D] expression for onDeviceFound:55 - 5

[D] Device::addDevice:128 - "Last device added: HXM037426"
[D] expression for onDeviceFound:55 - 6

[D] Device::addDevice:128 - "Last device added: 5A-4E-DA-62-9A-04"
[D] expression for onDeviceFound:55 - 7

[D] Device::addDevice:128 - "Last device added: 7E-F3-98-84-60-57"
[D] expression for onDeviceFound:55 - 8

[D] Device::deviceScanFinished:141 - Scan finished!
[D] Device::scanServices:157 - Trying to connect to  "00:07:80:0F:AC:95"
[D] Device::scanServices:185 - Disconnecting from previous random address device...
[D] Device::scanServices:193 - Trying to connect to address  "HXM037426"


[W] unknown:0 - void QBluetoothSocketPrivate::_q_readNotify() 58 error: -1 "Resource temporarily unavailable"
[W] Device::deviceDisconnected:346 - Disconnect from device

**Comment by thmichel** [comment]
Hi,

I think I've located the issue:

First we see it tries to connect to the correct device:
[D] Device::scanServices:157 - Trying to connect to "00:07:80:0F:AC:95"
Then it tries to disconnect an existing connection first - This should not happe as we do not have a connection!
[D] Device::scanServices:185 - Disconnecting from previous random address device...

Then we try to connect using classic Bluetooth, but with name instead of Address:
[D] Device::scanServices:193 - Trying to connect to address "HXM037426"

This fails of course. For a workaround, I commented out the disconnect section, now I can see a correct connection attempt.
[D] Device::scanServices:157 - Trying to connect to  "08:DF:1F:87:EB:01"
[D] Device::connectClassic:225 - Trying to connect to: 08:DF:1F:87:EB:01_1
[D] Device::connectClassic:237 - Connecting...
[D] Device::error:316 - Error:  "Host is down"

This error is expected - the device is switched off (can't test with HRM device as I don't have one).
So it should be fixable. I will look why the disconnect is being tried and will upload a new version tomorrow.

**Comment by thmichel** [comment]
Think I found it. When loading the Bluetooth settings page, the bluetooth type is not initialized so defaults to BLE Public. Only after switching to a different connection mode and back it works.
Adding         

id_Device.setBluetoothType(settings.bluetoothType);

in the onCompleted section should fix the error. I will upload a new version now.

**Comment by jdrescher2006** [comment]
You got it, it's working now!
Battery value with classic Bluetooth is working but not with BLE. Is this working for your BLE device?
And can we somehow avoid having 2 BLE entries in the combobox (BLE Public and BLE Random)?

**Comment by thmichel** [comment]
Hi,

good to hear it's working now. Sadly reducing it down to two items is not a good option (from my point of view). The app does not know if it's public or random type - you cannot detect this during scanning. So the only way would be to try first public, then random - that would cause a 30 seconds connection delay for random devices (or vice versa) and throw an error.

On another thing, I was thinking to put a shortcut to the bluetooth connection page into the main menu (bypassing the step to first go into the settings menu) as this probably is often used when starting the app - what to you think?

If you are happy with the current state, I would create a pull request so it can be merged into your code.

**Comment by jdrescher2006** [comment]
I wonder how the original bluetooth search of the operating system handles this. It lasts relatively long to detect my HR device so there might also be a 30 seconds delay. I am thinking about if a 30 seconds delay might be OK for us.

Why do you think it is neccessary to open up the bluetooth conection page? Only if bluetooth is deactivated?

Maybe you can prepare a further test version with a fix for the battery issue?

**Comment by thmichel** [comment]
Hi,

regarding random and public addreses this is what the QT documentation says:

_Note: All other platforms handle this flag transparently and therefore applications can ignore it entirely. On Linux, the address type flag is not directly exposed by BlueZ although some use cases do require this information. The only way to detect the flag is via the Linux kernel's Bluetooth Management API (kernel version 3.4+ required). This API requires CAP_NET_ADMIN capabilities though._

So it seems we cannot detect which type it is. I would like to avoid the delay as using the correct type lets it connect within 2-3 seconds. It's also no problem to hit the "Connect" button during scan given teh device has already been discovered (which usually happens fast).

The bluetooth connection page I was referring to was teh one from Laufhelden. Idea was to create a shortcut to it bypassing the settings menu to allow quick access to it as it probably is being used each time you start up Laufhelden.

I will create a new version this week where I try to fix the battery level issue - or at least get some better debug output for it.

**Comment by jdrescher2006** [comment]
Ok so let's stay with the three choices in the combobox.

Why would it be neccessary to go to the bluetooth page each time you start Laufhelden? I choose the HR device once and that's it. Is it not possible to save the selected bluetooth device to the settings and load it next time the app starts?

**Comment by thmichel** [comment]
The setting is saved and loaded on app start, but currently there is no automatic connect at startup. So to actually connect to the device, you have to first go to settings, heart rate device which takes you to the Bluetooth settings page where you have to hit "connect". I just wanted to speed that up a bit :-)

Another option would be to automatically try to connect to the saved device, or make that option a setting.

**Comment by jdrescher2006** [comment]
But there is already a mechanism to connect to the bluetooth device if a workout is started.
If the RecordPage is opened there are the lines:
if (sHRMAddress !== "" && settings.useHRMdevice && bRecordDialogRequestHRM === false && !settings.useHRMservice)
            {
                id_BluetoothData.connect(sHRMAddress, 1);
                bRecordDialogRequestHRM = true;
            }

Also check harbour-laufhelden.qml, there are other parts of the mechanism.

**Comment by thmichel** [comment]
Ah, thanks for pointing that out. That does not work for BLE devices - you need a scan first to discover them, just connecting to a known MAC does not work. So another thing to look at :-)

**Comment by thmichel** [comment]
OK, I have uploaded a new version that polls battery status every 10 seconds instead of waiting for the device to emit a signal. Please try if it helps.
I fixed the issue for connection to HRM at start of workout, tested only with BLE so far.

**Comment by jdrescher2006** [comment]
Sounds good, I will test it. Problem is I have currently no access to my BLE HR device (left it at my workplace). So I can only test bluetooth classic.

**Comment by ncartron** [comment]
Hi Jens and Tom,

I can test that today or tomorrow with my Polar BLE HR device.

Cheers,
Nico

**Comment by jdrescher2006** [comment]
@DrYak I hope we can merge it very soon, we just need to sort out a few thinks... I am planning a little christmas release :1st_place_medal: 

@ncartron the more testers we have, the better!

@thmichel I made some tests with your current version (bluetooth classic only). Here are the results:
- scanning starts when dialog opens. This is not good. User wants to choose connection type first and then start scanning manually.
- when on BTConnectPage.qml, after connecting to a classic HR device and then pressing disconnect button, nothing happens. No disconnect. Also combobox should then be invisible, it can be made visible if disconnection was successful.
- if dialog is left (BTConnectPage.qml), there is an error: nknown:38 - file:///usr/share/harbour-laufhelden/qml/pages/BTConnectPage.qml:38: TypeError: Property 'disconnect' of object Device(0xab80a8b8) is not a function
- After a workout, when the RecordPage is left, there is an error. After that the bluetooth device is not disconnected: [W] unknown:91 - file:///usr/share/harbour-laufhelden/qml/pages/PreRecordPage.qml:91: ReferenceError: id_BluetoothData is not defined
[D] Device::readData:549 - Entering readData...
[D] Device::readData:549 - Entering readData...
[D] Device::readData:549 - Entering readData...

**Comment by thmichel** [comment]
Hi,
the issues should be fixed; a new version is uploaded. The device disconnects now when the settings pages is closed (as it was before) and reconnects when starting a workout.
Currently the combobox in the settings page is always visible, hiding it seemed to disturb the UI feeling for my taste - can change that if you wish.

@ncartron & @DrYak : Do you see the battery status of the device as well?

**Comment by DrYak** [comment]
Battery status shows 100% on a fresh coin cell.

(On the other hand, these are coin cells, they [tend to have pretty voltage](https://electronics.stackexchange.com/a/234028) until almost right [at the end of life](http://www.arcadecollecting.com/dead/dead.html), so I'm not sure if that peculiar HRM model will ever tell something else beside "100%" and dead)

**Comment by ncartron** [comment]
Right, so I did run today with Laufhelden and my Polar HRM, and the HRM got disconnected pretty quickly while running, see enclosed picture :(

I'm pretty sure this is not because of the HRM, since it works well when i run with my Suunto watch.

On Tuesday, 18 December 2018, thmichel wrote:
> Hi,
> the issues should be fixed; a new version is uploaded. The device disconnects now when the settings pages is closed (as it was before) and reconnects when starting a workout.
> Currently the combobox in the settings page is always visible, hiding it seemed to disturb the UI feeling for my taste - can change that if you wish.
> 
> @ncartron & @DrYak : Do you see the battery status of the device as well? 
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/jdrescher2006/Laufhelden/issues/123#issuecomment-44836453

-- 
Sent from my Sailfish device

**Comment by thmichel** [comment]
Can you test with latest version and start laufhelden from a terminal to get the debug output? That would hopefully show the reason why it disconnects.

**Comment by jdrescher2006** [comment]
I had these disconnect issues with my classic bluetooth implementation. Therefore I implemented a timer to  constantly try to reconnect to the HR device. You can even restart Bluetooth on your smartphone while the workout, the timer will still try to reconnect.

**Comment by jdrescher2006** [comment]
@thmichel bluetoth classic now working properly, very good work! If you like, you can create a pull request.

**Comment by thmichel** [comment]
@jdrescher2006 : Pull request is created.
@ncartron : The reconnect timer wasn't working with 1.1.4-6, it should be working with 1.1.4-7.

**Comment by ncartron** [comment]
Just realised the attachment from Laufhelden was missing in my message from yesterday, so here we go:

![Uploading 20181219181102.jpg…]()

**Comment by ncartron** [comment]
> Can you test with latest version and start laufhelden from a terminal to get the debug output? That would hopefully show the reason why it disconnects.

Tom, maybe silly question, but how do I start laufhelden from a terminal? :)

**Comment by thmichel** [comment]
There is no such thing as a silly question :-)
You have to have the terminal app installed (Fingerterm). In there you just type harbour-laufhelden and hit enter.
Laufhelden opens and you can switch between the apps to see the terminal output. You also might want to update to the latest version as the periodic reconnect should work with this one.

**Comment by ncartron** [comment]
Thanks Tom!
OK, so I'll couple that with screen, since that'll be easier to grab the
output using a SSH session from my laptop :)

As I mentioned, I already have Laufhelden latest version:

[nemo@Sailfish ~]$ rpm -qa |grep -i lauf
harbour-laufhelden-1.1.4-7.armv7hl

Should be ok, correct?

On 20-Dec-2018 10:49 CET, <notifications@github.com> wrote:

> There is no such thing as a silly question :-)
> You have to have the terminal app installed (Fingerterm). In there you just type harbour-laufhelden and hit enter.
> Laufhelden opens and you can switch between the apps to see the terminal output. You also might want to update to the latest version as the periodic reconnect should work with this one.
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/jdrescher2006/Laufhelden/issues/123#issuecomment-448937778

-- 
Nico

**Comment by ncartron** [comment]
@thmichel ok i did workout today, and for some reason it worked perfectly :)

Here's the output (thanks screen! ;)
Note that I have removed the devices discovered.

Also, version is reported as 1.1.4-5, even though I have installed the latest RPM (1.1.4-7), as rpm -qa confirms.

[nemo@Sailfish ~]$ harbour-laufhelden
[D] unknown:0 - QML debugging is enabled. Only use this in a safe environment.
[D] main:55 - "Laufhelden"  version  "1.1.4-5"
[D] unknown:0 - Using Wayland-EGL
[D] HistoryModel::HistoryModel:69 - HistoryModel constructor
[D] Device::startDeviceDiscovery:105 - Scanning for devices ...
[D] Device::addDevice:128 - "<removed by Nico>"
[D] TrackRecorder::TrackRecorder:31 - TrackRecorder constructor
[D] TrackRecorder::loadAutoSave:959 - No autosave found
[D] TrackRecorder::TrackRecorder:62 - GPS initialized!
[D] TrackRecorder::setUpdateInterval:808 - Setting update interval to 1000 msec
[D] O2ReplyServer::O2ReplyServer:19 - O2ReplyServer: Starting
[D] O0BaseAuth::linked:33 - O0BaseAuth::linked: Yes
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:32:26 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::recordPagePortrait
[D] onStatusChanged:165 - First Active MainPage
[D] onStatusChanged:190 - sPebbleList: /org/rockwork/B0_B4_48_F7_0E_9F
[D] HistoryModel::loadAccelerationFile:342 - Reading helper file: "/home/nemo/Laufhelden/AccelerationHelper.dat"
[D] onStatusChanged:241 - Active MainPage
[D] TrackRecorder::vEndGPS:97 - Stopping GPS
[D] onStatusChanged:254 - vMainPageObject: MainPage_QMLTYPE_56(0xab4a4e28)
[D] expression for onSigAmountGPXFiles:412 - Amount of tracks: 10
[D] loadTrack:29 - Loading "running-Thu Nov 15 16:46:14 2018-11.2km.gpx"
[D] loadTrack:29 - Loading "Running-20181127-1159.gpx"
[D] loadTrack:33 - Already has data: "running-Thu Nov 15 16:46:14 2018-11.2km.gpx"
[D] loadTrack:33 - Already has data: "Running-20181127-1159.gpx"
[D] loadTrack:29 - Loading "other 1-Mon Nov 11 21:33:06 2013-27.3km.gpx"
[D] loadTrack:33 - Already has data: "other 1-Mon Nov 11 21:33:06 2013-27.3km.gpx"
[D] loadTrack:29 - Loading "Running-20181130-1600.gpx"
[D] loadTrack:29 - Loading "Running-20181129-1204.gpx"
[D] loadTrack:33 - Already has data: "Running-20181130-1600.gpx"
[D] loadTrack:33 - Already has data: "Running-20181129-1204.gpx"
[D] loadTrack:29 - Loading "Running-20181202-1513.gpx"
[D] loadTrack:29 - Loading "Running-20181204-0708.gpx"
[D] loadTrack:33 - Already has data: "Running-20181204-0708.gpx"
[D] loadTrack:29 - Loading "Running-20181213-1227.gpx"
[D] loadTrack:33 - Already has data: "Running-20181213-1227.gpx"
[D] loadTrack:29 - Loading "running-Wed Dec 19 15:59:33 2018-10.2km.gpx"
[D] loadTrack:29 - Loading "Running-20181219-1559.gpx"
[D] loadTrack:33 - Already has data: "running-Wed Dec 19 15:59:33 2018-10.2km.gpx"
[D] loadTrack:33 - Already has data: "Running-20181219-1559.gpx"
[D] loadTrack:33 - Already has data: "Running-20181202-1513.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "running-Thu Nov 15 16:46:14 2018-11.2km.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "other 1-Mon Nov 11 21:33:06 2013-27.3km.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "Running-20181130-1600.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "Running-20181129-1204.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "Running-20181204-0708.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "Running-20181127-1159.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "Running-20181213-1227.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "running-Wed Dec 19 15:59:33 2018-10.2km.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "Running-20181219-1559.gpx"
[D] HistoryModel::newTrackData:300 - Finished loading "Running-20181202-1513.gpx"
[D] HistoryModel::loadingFinished:309 - Data loading finished
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:334 - file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:334: TypeError: Cannot read property 'iDistance' of undefined
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:766 - file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:766: TypeError: Cannot read property 'icon' of undefined
[D] Device::deviceScanFinished:141 - Scan finished!
[D] onApplicationActiveChanged:33 - onApplicationActiveChanged: true
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/BTConnectPage.qml:12:34 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::bluetoothType
[D] Device::setBluetoothType:556 - Set Bluetooth type to  0
[D] Device::setBluetoothType:556 - Set Bluetooth type to  0
[D] onStatusChanged:28 - BTPage
[D] Device::setBluetoothType:556 - Set Bluetooth type to  0
[D] Device::scanServices:157 - Trying to connect to  "00:22:D0:95:F7:9E"
[D] Device::scanServices:175 - Device to connect not found!
[D] Device::scanServices:157 - Trying to connect to  "00:22:D0:95:F7:9E"
[D] Device::scanServices:175 - Device to connect not found!
[D] Device::startDeviceDiscovery:105 - Scanning for devices ...
[D] Device::addDevice:128 - "Last device added: Bose SoundSport"
[D] expression for onDeviceFound:58 - 1
[D] Device::addDevice:128 - "Last device added: Bose Solo 5 system"
[D] expression for onDeviceFound:58 - 2
[D] Device::addDevice:128 - "Last device added: Nico?s AirPods"
[D] expression for onDeviceFound:58 - 3
[D] Device::addDevice:128 - "Last device added: NISSAN CONNECT"
[D] expression for onDeviceFound:58 - 4
[D] Device::addDevice:128 - "Last device added: Pebble Time 0E9F"
[D] expression for onDeviceFound:58 - 5
[D] Device::addDevice:128 - "Last device added: Amazfit Bip Watch"
[D] expression for onDeviceFound:58 - 6
[D] Device::addDevice:128 - "Last device added: 55-FA-FC-2B-6F-08"
[D] expression for onDeviceFound:58 - 7
[D] Device::addDevice:128 - "Last device added: Polar H7 95F79E10"
[D] expression for onDeviceFound:58 - 8
[D] Device::scanServices:157 - Trying to connect to  "00:22:D0:95:F7:9E"
[D] Device::scanServices:199 - Trying to connect to BLE address  "Polar H7 95F79E10"
[D] Device::deviceConnected:303 - BLE Device connected - Discovering services...)
[W] unknown:0 - Cannot find matching characteristic for notification/indication
[D] Device::lowEnergyServiceDiscovered:251 - Heart Rate Monitor Found
[D] Device::lowEnergyServiceDiscovered:256 - Battery Level Found
[D] Device::serviceScanDone:273 - Service scan done!)
[D] Device::hrmServiceStateChanged:389 - Discovering services...
[D] Device::batServiceStateChanged:430 - Discovering services...
[D] Device::batServiceStateChanged:434 - Service discovered.
[D] Device::subscribeToHRM:418 - Subscribing to HRM Service
[D] Device::addDevice:128 - "Last device added: Freebox Player Mini v2"
[D] expression for onDeviceFound:58 - 9
[D] Device::deviceScanFinished:141 - Scan finished!
[D] Device::setBluetoothType:556 - Set Bluetooth type to  0
[W] Device::deviceDisconnected:372 - Disconnect from device
[D] Device::setBluetoothType:556 - Set Bluetooth type to  0
[D] onStatusChanged:241 - Active MainPage
[D] TrackRecorder::vEndGPS:97 - Stopping GPS
[D] onStatusChanged:254 - vMainPageObject: MainPage_QMLTYPE_56(0xab4a4e28)
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/PreRecordPage.qml:27:26 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::recordPagePortrait
[D] onStatusChanged:40 - First Active PreRecordPage
[D] onStatusChanged:43 - Index of workout type: 0
[D] onStatusChanged:67 - arrayThresholdProfiles.length: 3
[D] onStatusChanged:68 - selected profile index: 0
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:2206:35 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapCache
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:2209:19 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapStyle
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:70:42 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:48:32 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::displayMode
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:2020:26 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:49:37 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::showBorderLines
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:69:42 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:1920:22 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:1809:22 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:68:42 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:1133:26 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::enablePebble
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:1123:26 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::useHRMdevice
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:1113:26 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::useHRMdevice
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:83:32 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapDisableRecordPage
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:34:26 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::recordPagePortrait
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:48:32 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::displayMode
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:49:37 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::showBorderLines
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:65:35 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::displayMode
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:68:42 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:69:42 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:70:42 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapShowOnly4Fields
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/RecordPage.qml:83:32 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapDisableRecordPage
[D] onCompleted:103 - dbus completed
[D] onStatusChanged:83 - PreRecordPage active
[W] unknown:0 - [ INFO ]  "{QSGRenderThread}[General]: GPU Identifier: Adreno (TM) 510"
[D] onCurrentItemChanged:219 - Selected profile: Thresholds off
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:810:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:793:31 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:738:38 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:766 - file:///usr/share/harbour-laufhelden/qml/pages/MainPage.qml:766: TypeError: Cannot read property 'icon' of undefined
[D] O2ReplyServer::O2ReplyServer:19 - O2ReplyServer: Starting
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:369:24 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:363:24 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:358:24 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[D] O0BaseAuth::linked:33 - O0BaseAuth::linked: Yes
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:251:26 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::stUsername
[W] unknown:174 - file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:174: TypeError: Cannot read property 'icon' of undefined
[W] unknown:185 - file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:185: TypeError: Cannot read property 'labeltext' of undefined
[D] TrackLoader::setFilename:680 - Setting filename "running-Fri Dec 21 12:09:00 2018-10.8km.gpx"
[D] TrackLoader::load:579 - m_distance first calculated:  "10846.8"
[D] TrackLoader::load:649 - m_distance second calculated:  "10846.8"
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:369:24 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:363:24 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:358:24 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/DetailedViewPage.qml:363:24 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::measureSystem
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MapViewPage.qml:210:35 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapCache
[W] unknown:0 - QQmlExpression: Expression file:///usr/share/harbour-laufhelden/qml/pages/MapViewPage.qml:213:19 depends on non-NOTIFYable properties:
[W] unknown:0 -     Settings::mapStyle
[D] update:422 - dist: 20000
[D] onApplicationActiveChanged:33 - onApplicationActiveChanged: false
[W] unknown:0 - void QBluetoothSocketPrivate::_q_readNotify() 66 error: -1 "Connection reset by peer"
[W] Device::deviceDisconnected:372 - Disconnect from device
[D] onApplicationActiveChanged:33 - onApplicationActiveChanged: true
[D] onApplicationActiveChanged:33 - onApplicationActiveChanged: false

**Comment by jdrescher2006** [comment]
Hi guys, could someone test the current version in my repo with a BLE device? I am still working on a few bugs but would like to release this version in a few days.
I can also provide a RPM if requested.

**Comment by thmichel** [comment]
@ncartron : Glad to hear it works for you now. I merged the latest changes from Jens and uploaded an RPM file to my fork. Only change is version number so that it's not the same as the current release. You may want to test with this one until Jens has uploaded an RPM.

Will do some testing myself tomorrow.

**Comment by jdrescher2006** [comment]
@thmichel thanks for your upload! I have now also uploaded an RPM:
[https://drive.google.com/open?id=1H5ioct5691taz8A_iKfoNpmLPeJFo3Ya](url)
Made some minor changes and bugfixes, will push this soon.

**Comment by ncartron** [comment]
Hi Jens,

filename is harbour-laufhelden-1.1.3-1.armv7hl.rpm, is that normal?
I was expecting a version of at least 1.1.4-*?

Nico

On 21-Dec-2018 20:45 CET, <notifications@github.com> wrote:

> @thmichel thanks for your upload! I have now also uploaded an RPM:
> [https://drive.google.com/open?id=1H5ioct5691taz8A_iKfoNpmLPeJFo3Ya](url)
> Made some minor changes and bugfixes, will push this soon.
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/jdrescher2006/Laufhelden/issues/123#issuecomment-449481555

-- 
Nico

**Comment by ncartron** [comment]
Just realized that current version on Openrepos is 1.1.2-1, so having the RPM with 1.1.3-1 makes sense.

I'll use Jens' RPM if you don't mind Tom, since it seems there are also other changes.

Will workout on Sunday so will let you know how it goes! 

On Friday, 21 December 2018, Jens Drescher wrote:
> @thmichel thanks for your upload! I have now also uploaded an RPM:
> [https://drive.google.com/open?id=1H5ioct5691taz8A_iKfoNpmLPeJFo3Ya](url)
> Made some minor changes and bugfixes, will push this soon.
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/jdrescher2006/Laufhelden/issues/123#issuecomment-44948155

-- 
Sent from my Sailfish device

**Comment by thmichel** [comment]
Sure, will do the same :-)
Just uploaded it in case Jens would not find the time. And 1.1.3  will be good, just used 1.1.4 to differentiate from Jens' files.

**Comment by thmichel** [comment]
Did some basic tests so far with 1.1.3-1. Looks good so far, connect and disconnect works fin, battery status is working too.

**Comment by jdrescher2006** [comment]
Yes, I did a workout today and everything was fine (bluetooth classic).

**Comment by ncartron** [comment]
Same here, did a longer run (17K) today and 1.1.3-1 worked perfectly with BLE.
I tested fully with Pebble, voice coach and bluetooth headset with Spotify running, no problem at all, no disconnection.

I still don't have battery status but again j don't remember seeing it with the official Polar app.

On Sunday, 23 December 2018, Jens Drescher wrote:
> Yes, I did a workout today and everything was fine (bluetooth classic).
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/jdrescher2006/Laufhelden/issues/123#issuecomment-44965607

-- 
Sent from my Sailfish device

**Comment by jdrescher2006** [comment]
Update out, really appreciate working with you guys!!!

**Comment by DrYak** [comment]
Currently testing while skiing during my end-of-year vaccations:
 - BLE still works nicely

Also, tested with a discharged battery (our kitchen scale was refusing to use it):
 - App reports only 90% battery left
(Which, again, is expected on coin cells, they have near to max voltage until right about to die. 90% is the expected result)

**Comment by thmichel** [comment]
@DrYak Great to hear it works well for you. Seems jdrescher can close the issue now.

Next thing I'ld like to see implemented is workouts without GPS (e.g. cycling in the gym). So just collect time and hart rate to have some statistics. Something to look at next year probably :-)

**Comment by jdrescher2006** [comment]
@thmichel workouts without GPS would be really cool! Lots of work to do though...
